rest_framework/filters.py

```diff
@@ -80,67 +40,33 @@ class DjangoFilterBackend(BaseFilterBackend):
     """
     A filter backend that uses django-filter.
     """
-    default_filter_set = FilterSet
-    template = filter_template
-
-    def __init__(self):
+    def __new__(cls, *args, **kwargs):
         assert django_filters, 'Using DjangoFilterBackend, but django-filter is not installed'
+        assert django_filters.VERSION >= (0, 15, 0), 'django-filter 0.15.0 and above is required'
 
-    def get_filter_class(self, view, queryset=None):
-        """
-        Return the django-filters `FilterSet` used to filter the queryset.
-        """
-        filter_class = getattr(view, 'filter_class', None)
-        filter_fields = getattr(view, 'filter_fields', None)
-
-        if filter_class:
-            filter_model = filter_class.Meta.model
+        warnings.warn(
+            "'rest_framework.filters.DjangoFilterBackend' has been deprecated "
+            "in favor of 'django_filters.rest_framework.DjangoFilterBackend'",
+            DeprecationWarning
+        )
 
-            assert issubclass(queryset.model, filter_model), \
-                'FilterSet model %s does not match queryset model %s' % \
-                (filter_model, queryset.model)
+        from django_filters.rest_framework import DjangoFilterBackend as DFBackend
 
-            return filter_class
+        class DjangoFilterDeprecationBackend(DFBackend):
 
-        if filter_fields:
-            class AutoFilterSet(self.default_filter_set):
-                class Meta:
-                    model = queryset.model
-                    fields = filter_fields
+            # get_fields was introduced after the move to django-filter
+            def get_fields(self, view):
```

Comment:
Given my comment [here](https://github.com/carltongibson/django-filter/pull/492#discussion_r79999249), I'm not really sure how to write a meaningful test.